internal/scm/azuredevops/repository.go

```diff
@@ -0,0 +1,51 @@
+package azuredevops
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+)
+
+func (a *AzureDevOps) convertRepo(repo *git.GitRepository) repository {
+	var cloneURL string
+	if a.SSHAuth {
+		cloneURL = *repo.SshUrl
+	} else {
+		cloneURL = fmt.Sprintf("https://%s@%s", a.pat, strings.TrimPrefix(*repo.RemoteUrl, "https://"))
```

Comment:
I must say I prefer this method. But I leave it up to you if you want to switch or not
```suggestion
		u, err := url.Parse(*repo.RemoteUrl)
		if err != nil {
			return repository{}, errors.Wrap(err, "could not parse Azure Devops remote url")
		}
		// Set the token as https://ACCESSTOKEN@remote-url
		u.User = url.User(a.pat)
		cloneURL = u.String()
```

Reply:
I'll definitely use this.  I Didn't know it was built in like this!